src/rgw/rgw_dedup.cc

```diff
@@ -531,12 +546,17 @@ namespace rgw::dedup {
                           << obj << dendl;
         continue;
       }
-      librados::IoCtx ioctx = obj.ioctx;
-      ldpp_dout(dpp, 20) << __func__ << "::removing tail object: " << raw_obj.oid
-                         << dendl;
-      ret = ioctx.remove(raw_obj.oid);
+      ldpp_dout(dpp, 20) << __func__ << "::removing tail object: "
+                         << raw_obj.oid << dendl;
+      d_ctl.metadata_access_throttle.acquire();
+      ObjectWriteOperation op;
+      rgw::AioResultList completed;
+      cls_refcount_put(op, ref_tag, true);
```

Comment:
why do we need `cls_refcount` for throttling?

Reply:
No, this is an issue I found in the code while making the throttling change.
My code was using brute-force remove for the tail-objects instead of ref_count put which could lead to premature deletes if the tail-object was linked to another head-object (after a server-side-copy).

I could have used a standalone PR for this, but it takes so long to get a PR in with teuthology failures, build failures etc...